src/components/component.ts

```diff
@@ -450,10 +450,10 @@ export class Component {
 
     if (this._isAnchored) {
       this._element.remove();
+      this._isAnchored = false;
+      this._onDetachCallbacks.callCallbacks(this);
```

Comment:
Can you explain why these changes should be in this pull request?  It is not that clear from the description why this is part of the fix.  Also, I feel that we should ask if `detach()` on un-anchored components should call the callbacks.